core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala

```diff
@@ -0,0 +1,138 @@
+  def YARNQueue = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_YARN_QUEUE._1)
+  def h2oDriverPath = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_H2O_DRIVER._1)
+  def numOfExternalH2ONodes = sparkConf.getOption(PROP_EXTERNAL_H2O_NODES._1)
+  def HDFSOutputDir = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_HDFS_DIR._1)
+  def mapperXmx = sparkConf.get(PROP_EXTERNAL_H2O_MEMORY._1, PROP_EXTERNAL_H2O_MEMORY._2)
+
+
+  /**
+    * Sets node and port representing H2O Cluster to which should H2O connect when started in external mode.
+    * This method automatically sets external cluster mode
+    *
+    * @param host host representing the cluster
+    * @param port port representing the cluster
+    * @return H2O Configuration
+    */
+  def setH2OCluster(host: String, port: Int): H2OConf = {
+    setExternalClusterMode()
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1, host + ":" + port)
+    self
+  }
+
+  def setH2OCluster(hostPort: String): H2OConf = {
+    setExternalClusterMode()
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1, hostPort)
+    self
+  }
+
+  def h2oClusterHost = {
+    val value = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1)
+    if (value.isDefined) {
+      Option(value.get.split(":")(0))
+    } else {
+      None
+    }
+  }
+
+  def h2oClusterPort = {
+    val value = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1)
+    if (value.isDefined) {
+      Option(value.get.split(":")(1))
+    } else {
+      None
+    }
+  }
+
+
+  def setYARNQueue(queueName: String) = {
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_YARN_QUEUE._1, queueName)
+    self
+  }
+
+  def setH2ODriverPath(path: String): H2OConf = {
+    setExternalClusterMode()
+    logWarning("Using external cluster mode")
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_H2O_DRIVER._1, path)
+    self
+  }
+
+  def setNumOfExternalH2ONodes(numOfExternalH2ONodes: Int): H2OConf = {
+    sparkConf.set(PROP_EXTERNAL_H2O_NODES._1, numOfExternalH2ONodes.toString)
+    self
+  }
+
+  def setHDFSOutputDir(dir: String): H2OConf = {
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_HDFS_DIR._1, dir)
+    self
+  }
+
+
+  def setMapperXmx(mem: String): H2OConf = {
+    sparkConf.set(PROP_EXTERNAL_H2O_MEMORY._1, mem)
+    self
+  }
+
+
+  def externalConfString: String =
+    s"""Sparkling Water configuration:
+        |  backend cluster mode : ${backendClusterMode}
+        |  cloudName            : ${cloudName.getOrElse("Not set yet")}
+        |  cloud representative : ${h2oCluster.getOrElse("Not set, using cloud name only")}
+        |  clientBasePort       : ${clientBasePort}
+        |  h2oClientLog         : ${h2oClientLogLevel}
+        |  nthreads             : ${nthreads}""".stripMargin
+}
+
+object ExternalBackendConf {
+
+  val PROP_EXTERNAL_H2O_MEMORY = ("spark.ext.h2o.hadoop.memory", "6g")
+
+  val PROP_EXTERNAL_CLUSTER_HDFS_DIR = ("spark.ext.h2o.external.hdfs.dir", null.asInstanceOf[String])
```

Comment:
I would like to use null for the sake of consistency with rest of the configuration properties at least as part of this change. I'm open up to eliminating `null`s as part of another commit targeting particularly this change.